deltachat-ffi/deltachat.h

```diff
@@ -183,19 +183,70 @@ typedef struct _dc_accounts_event_emitter dc_accounts_event_emitter_t;
  * opened, connected and mails are fetched.
  *
  * @memberof dc_context_t
- * @param os_name is only for decorative use.
- *     You can give the name of the app, the operating system,
- *     the used environment and/or the version here.
  * @param dbfile The file to use to store the database,
  *     something like `~/file` won't work, use absolute paths.
- * @param blobdir Deprecated, pass NULL or an empty string here.
  * @return A context object with some public members.
  *     The object must be passed to the other context functions
  *     and must be freed using dc_context_unref() after usage.
  *
  * If you want to use multiple context objects at the same time,
  * this can be managed using dc_accounts_t.
  */
+dc_context_t*   dc_context_new_closed        (const char* dbfile);
+
+
+/**
+ * Sets the database passphrase. This can only be used on closed context, such
+ * as created by dc_context_new_closed(). If the database is new, this
+ * operation sets the database passphrase. For existing databases the
+ * passphrase should be the one used to encrypt the database the first time.
```

Comment:
what happens if used on an existing context that didn't have a password set at creation time?

I kind of feel like this could be a bit hard to use.  Could we have a `dc_context_new_with_password()` to improve this?  Maybe with a `dc_db_requires_password()` so the UI can figure out if it is needed or not?  I apologise if I missed or forgot some discussion of why this wouldn't be good.

Reply:
i think, that recent discussion from irc is related:
```
-- Fri, 07 Jan 2022 --
r10s: why not just add the passphrase to dc_context_new()?
link2xt: then what will account manager return as a context that is not open yet because passphrase was not provided?
link2xt: my idea is that it returns a "closed context"
link2xt: you then set passphrase and open it
link2xt: and after that start_io will work on it
```
so, imu, the thing is that we need sth. the account manager can return.

for m1 it is probably not required for ui to get know if a given context requires a passphrase or not, the ui will know as it has stored it somewhere, so, ui will know if they have to call `dc_context_set_passphrase()` or not. but yes, it is a good question what happens when ui does not know if a passphrase is required.

Reply:
Account manager opens and starts all passwordless contexts automatically. After that, you go through accounts and check if they are open already with dc_context_is_open().

For closed contexts, you can check if they don't need the password with `dc_context_set_password(context, NULL)`. If the context is already open, you can't close them and don't need to care which password they have, either use the context or unref it.

Reply:
this still leaves some guesswork does it not?  would `dc_open_with_passphrase()` be a better alternative to opening a database that needs a passphrase?  this function is then only to set a passphrase on a (new) account and the semantics of it become simpler

Reply:
We can then remove `dc_context_new_closed()` and just open legacy contexts, while for encrypted database you need to call `dc_context_is_open()` and then `dc_context_open(context, "passphrase")`. Less stateful API is generally better, will try to add it on top as a separate commit.

Reply:
I pushed a second commit removing `dc_context_set_passphrase()`.